workflow/scripts/ts_processors.py

```diff
@@ -198,110 +198,63 @@ class tskit_windowed_sfs_plus_ld(BaseProcessor):
         selected_snps = []
         for gap in gaps:
             snps = np.arange(0, n_SNP, gap) + np.random.randint(0, (n_SNP - 1) % gap + 1)
-            # adding a random start (+1, bc 2nd bound in randint is exlusive)
-
-            # non overlapping contiguous pairs
-            # snps=[ 196, 1220, 2244] becomes
-            # snp_pairs=[(196, 1220), (1221, 2245)]
-            snp_pairs = np.unique([((snps[i] + i) % n_SNP, (snps[i + 1] + i) % n_SNP) for i in range(len(snps) - 1)],
-                                  axis=0)
+            snp_pairs = np.unique([((snps[i] + i) % n_SNP, (snps[i + 1] + i) % n_SNP)
+                                for i in range(len(snps) - 1)], axis=0)
 
-            # If we don't have enough pairs (typically when gap is large), we add a random rotation until we have at
-            # least 300) count = 0
-
-            if not circular:
-                snp_pairs = snp_pairs[snp_pairs[:, 0] < snp_pairs[:, 1]]
+            snp_pairs = snp_pairs[snp_pairs[:, 0] < snp_pairs[:, 1]]
             last_pair = snp_pairs[-1]
-
-            if circular:
-                max_value = n_SNP - 1
-            else:
-                max_value = n_SNP - gap - 1
+            max_value = n_SNP - gap - 1
 
             while len(snp_pairs) <= min(300, max_value):
                 random_shift = np.random.randint(1, n_SNP) % n_SNP
                 new_pair = (last_pair + random_shift) % n_SNP
                 snp_pairs = np.unique(np.concatenate([snp_pairs, new_pair.reshape(1, 2)]), axis=0)
                 last_pair = new_pair
-
-                if not circular:
-                    snp_pairs = snp_pairs[snp_pairs[:, 0] < snp_pairs[:, 1]]
-
+                snp_pairs = snp_pairs[snp_pairs[:, 0] < snp_pairs[:, 1]]
             selected_snps.append(snp_pairs)
 
-        # Functions to aggregate the values within each distance bin
-        agg_bins = {"snp_dist": ["mean"], "r2": ["mean", "count", "sem"]}
+        # Collect r2 values into a DataFrame.
+        agg_bins = {"snp_dist": ["mean"], "r2": ["mean"]}
 
         ld = pd.DataFrame()
         for i, snps_pos in enumerate(selected_snps):
-
-            if circular:
-                sd = pd.DataFrame((np.diff(pos_vec[snps_pos]) % size_chr) % (size_chr // 2),
-                                  columns=["snp_dist"])  # %size_chr/2 because max distance btw 2 SNP is size_chr/2
-            else:
-                sd = pd.DataFrame((np.diff(pos_vec[snps_pos])), columns=["snp_dist"])
-
+            sd = pd.DataFrame((np.diff(pos_vec[snps_pos])), columns=["snp_dist"])
             sd["dist_group"] = pd.cut(sd.snp_dist, bins=distance_bins)
             sr = [allel.rogers_huff_r(snps) ** 2 for snps in haplotype[snps_pos]]
             sd["r2"] = sr
             sd["gap_id"] = i
             ld = pd.concat([ld, sd])
 
-        ld2 = ld.dropna().groupby("dist_group", observed=False).agg(agg_bins)
+        ld2 = ld.dropna().groupby("dist_group",observed=True).agg(agg_bins)
```

Comment:
i believe that setting `observed=True` here will throw an error, at least it did for me when first reimplementing this routine